apps/web/app/(app)/(survey-editor)/environments/[environmentId]/surveys/[surveyId]/edit/components/CardStylingSettings.tsx

```diff
@@ -194,107 +79,236 @@ export const CardStylingSettings = ({
+                </FormControl>
+              </FormItem>
+            )}
           />
 
-          <ColorSelectorWithLabel
-            label="Card shadow color"
-            color={cardShadowColor}
-            setColor={setCardShadowColor}
-            description="Change the shadow color of the card."
+          <FormField
+            control={form.control}
+            name="cardBorderColor.light"
+            render={({ field }) => (
+              <FormItem className="space-y-4">
+                <div>
+                  <FormLabel className="text-sm font-semibold text-slate-700">Card border color</FormLabel>
+                  <FormDescription>Change the border color of the card.</FormDescription>
+                </div>
+
+                <FormControl>
+                  <ColorSelectorWithLabel
+                    color={field.value || COLOR_DEFAULTS.cardBorderColor}
+                    setColor={(color) => field.onChange(color)}
+                  />
+                </FormControl>
+              </FormItem>
+            )}
           />
 
-          <CardArrangement
-            surveyType={isAppSurvey ? "app" : "link"}
-            activeCardArrangement={isAppSurvey ? inAppSurveyCardArrangement : linkSurveyCardArrangement}
-            setActiveCardArrangement={setCardArrangement}
+          <FormField
+            control={form.control}
+            name="cardShadowColor.light"
+            render={({ field }) => (
+              <FormItem className="space-y-4">
+                <div>
+                  <FormLabel className="text-sm font-semibold text-slate-700">Card shadow color</FormLabel>
+                  <FormDescription>Change the shadow color of the card.</FormDescription>
+                </div>
+
+                <FormControl>
+                  <ColorSelectorWithLabel
+                    color={field.value || COLOR_DEFAULTS.cardShadowColor}
+                    setColor={(color) => field.onChange(color)}
+                  />
+                </FormControl>
+              </FormItem>
+            )}
           />
 
-          <>
-            <div className="flex items-center space-x-1">
-              <Switch
-                id="hideProgressBar"
-                checked={!!hideProgressBar}
-                onCheckedChange={(checked) => toggleProgressBarVisibility(checked)}
-              />
-              <Label htmlFor="hideProgressBar" className="cursor-pointer">
-                <div className="ml-2">
-                  <h3 className="text-sm font-semibold text-slate-700">Hide progress bar</h3>
-                  <p className="text-xs font-normal text-slate-500">
-                    Disable the visibility of survey progress.
-                  </p>
+          <FormField
+            control={form.control}
+            name="cardArrangement"
+            render={({ field }) => (
+              <FormItem>
+                <div>
+                  <FormLabel className="text-sm font-semibold text-slate-700">
```

Comment:
Card arrangements is also not always updating in my testing:


https://github.com/formbricks/formbricks/assets/675065/53c5cbc1-264a-43f0-8c14-1629331a7a7e